src/lib/Permission/PermissionChecker.php

```diff
@@ -158,6 +190,27 @@ public function canCreateInLocation(Location $location, $hasAccess): bool
         return true;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getContentCreateLimitations(Location $parentLocation): LookupLimitationResult
+    {
+        $contentType = $this->contentTypeService->loadContentType($parentLocation->contentInfo->contentTypeId);
+        $contentCreateStruct = $this->contentService->newContentCreateStruct($contentType, $parentLocation->contentInfo->mainLanguageCode);
+        $contentCreateStruct->sectionId = $parentLocation->contentInfo->sectionId;
+        $locationCreateStruct = $this->locationService->newLocationCreateStruct($parentLocation->id);
+
+        $versionBuilder = new VersionBuilder();
+        $versionBuilder->translateToAnyLanguageOf($this->getActiveLanguageCodes());
+        $versionBuilder->createFromAnyContentTypeOf($this->getContentTypeIds());
```

Comment:
This gives me a feeling that we've missed use case that should be handled internally. `getContentTypeIds` is so heavy, it should rather be delegated to persistence handler and gateway which will return full list. Meaning: 
```php
$versionBuilder->createFromAnyContentType();
```
^ handled internally.

But huh, that's a lot of work probably. Maybe an optimization follow up?

Reply:
We could handle this optimization as a follow-up